File: common/common/eventbus/topics.py
from __future__ import annotations

from .core import Topic


TOPIC_POST_SUMMARY = Topic("tech-letter.post.summary")
TOPIC_POST_EMBEDDING = Topic("tech-letter.post.embedding")
TOPIC_CREDIT = Topic("tech-letter.credit")
TOPIC_CHAT = Topic("tech-letter.chat")

ALL_TOPICS: list[Topic] = [
    TOPIC_POST_SUMMARY,
    TOPIC_POST_EMBEDDING,
    TOPIC_CREDIT,
    TOPIC_CHAT,
]

File: cmd/internal/eventbus/topics.go
package eventbus

// 전역 토픽 선언: 기능별 기본 토픽 이름을 관리합니다.
// 필요시 환경설정으로 교체할 수 있도록 한 곳에서 관리합니다.

var (
	TopicPostEvents = NewTopic("tech-letter.post.events")
)

var AllTopics = []Topic{
	TopicPostEvents,
}
